### PR TITLE
data tokenization refactoring

### PR DESCRIPTION
**Pull request by Nuzhdina-Elena**
Refactoring moved from BeamFromDemo.
DataTokenizationTest fixed.

### CODE REVIEW

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/ProtegrityDataProtectors.java [hunk 1]
```diff
@@ -150,7 +150,7 @@ public PCollectionTuple expand(PCollection<KV<Integer, Row>> inputRows) {
     /**
      * Logger for class.
      */
-    private static final Logger LOG = LoggerFactory.getLogger(DSGTokenizationFn.class);
+    private static final Logger LOG = LoggerFactory.getLogger(BigQueryIO.class);
```
Comment (KhaninArtur) [inline v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/ProtegrityDataProtectors.java:153]: Seems like it's a wrong class for logger

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/ProtegrityDataProtectors.java [hunk 2]
```diff
@@ -298,12 +298,6 @@ private String formatJsonsToDsgBatch(Iterable<String> jsons) {
       CloseableHttpResponse response = sendToDsg(
           formatJsonsToDsgBatch(rowsToJsons(inputRows)).getBytes());
 
-      if (response.getStatusLine().getStatusCode() != HttpStatus.SC_OK) {
-        LOG.error("Send to DSG '{}' failed with '{}'",
-            this.dsgURI,
-            response.getStatusLine());
-      }
-
```
Comment (KhaninArtur) [inline v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/ProtegrityDataProtectors.java:306]: Probably, you should try rebasing on master - this is a recent fix from Mikhail, we shouldn't delete it 🙂

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/io/GcsIO.java [hunk 3]
```diff
@@ -173,123 +173,156 @@ public GcsIO(ProtegrityDataTokenizationOptions options) {
...
+                .via(AvroUtils.getGenericRecordToRowFunction(beamSchema)))
+        .setCoder(RowCoder.of(beamSchema));
+
+  }
+
+  private PCollectionTuple readCsv(Pipeline pipeline) {
+    return pipeline
         /*
-         * Step 3: Write jsons to dead-letter gcs that were successfully processed.
+         * Step 1: Read CSV file(s) from Cloud Storage using {@link CsvConverters.ReadCsv}.
```
Comment (KhaninArtur) [inline v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/io/GcsIO.java:198]: I think this comment is redundant

**Review by ilya-kozyrev — APPROVED** [review]
LGTM

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/io/GcsIO.java [hunk 4]
```diff
@@ -173,123 +173,156 @@ public GcsIO(ProtegrityDataTokenizationOptions options) {
...
+  private PCollection<Row> readAvro(Pipeline pipeline, Schema beamSchema) {
+    org.apache.avro.Schema avroSchema = AvroUtils.toAvroSchema(beamSchema);
+    PCollection<GenericRecord> genericRecords = pipeline.apply(
+        "ReadAvroFiles",
+        AvroIO.readGenericRecords(avroSchema).from(options.getInputGcsFilePattern()));
+    return genericRecords
+        .apply(
+            "GenericRecordToRow", MapElements.into(TypeDescriptor.of(Row.class))
+                .via(AvroUtils.getGenericRecordToRowFunction(beamSchema)))
+        .setCoder(RowCoder.of(beamSchema));
```
Comment (KhaninArtur) [inline v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/io/GcsIO.java:191]: Let's merge these 2 parts into 1 the same way it is done in `writeAvro` function.

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/io/GcsIO.java [hunk 5]
```diff
@@ -173,123 +173,156 @@ public GcsIO(ProtegrityDataTokenizationOptions options) {
...
+    switch (options.getOutputGcsFileFormat()) {
+      case JSON:
+        return writeJson(input);
+      case AVRO:
+        return writeAvro(input, schema);
+      case CSV:
+        return writeCsv(input, schema.getFieldNames());
+      default:
+        throw new IllegalStateException(
+            "No valid format for output data is provided. Please, choose JSON or CSV or AVRO.");
```
Comment (KhaninArtur) [inline v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/io/GcsIO.java:324]: Please, change the message to this:
"No valid format for output data is provided. Please, choose JSON, CSV, or AVRO."

File: v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/io/GcsIO.java [hunk 6]
```diff
@@ -173,123 +173,156 @@ public GcsIO(ProtegrityDataTokenizationOptions options) {
...
+
     }
   }
 }
```
Comment (KhaninArtur) [inline v2/protegrity-data-tokenization/src/main/java/com/google/cloud/teleport/v2/transforms/io/GcsIO.java:328]: Please, also don't delete log4j.properties file

**Pull request merged**